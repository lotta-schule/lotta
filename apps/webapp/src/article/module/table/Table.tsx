import * as React from 'react';
import { Edit } from './Edit';
import { Show } from './Show';
import { ContentModuleComponentProps } from '../ContentModule';

import styles from './Table.module.scss';

export interface TableCell {
  text: string;
}

export interface TableContent {
  rows: TableCell[][];
}

export type TableConfiguration = Record<never, never>; // Have done nothing yet

export const Table = React.memo(
  ({
    isEditModeEnabled,
    contentModule,
    onUpdateModule,
  }: ContentModuleComponentProps) => {
    return (
      <div className={styles.root} data-testid="TableContentModule">
        {isEditModeEnabled && onUpdateModule && (
          <Edit contentModule={contentModule} onUpdateModule={onUpdateModule} />
        )}
        {(!isEditModeEnabled || !onUpdateModule) && (
          <Show contentModule={contentModule} />
        )}
      </div>
    );
  }
);
Table.displayName = 'Table';
