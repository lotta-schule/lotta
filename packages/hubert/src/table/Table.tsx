import * as React from 'react';
import clsx from 'clsx';

import styles from './Table.module.scss';

export type TableProps = Omit<React.HTMLProps<HTMLTableElement>, 'ref'> & {
  ref?: React.ForwardedRef<HTMLTableElement>;
};

export const Table = React.forwardRef<HTMLTableElement, TableProps>(
  ({ children, className, ...props }, ref) => {
    return (
      <table ref={ref} className={clsx(styles.root, className)} {...props}>
        {children}
      </table>
    );
  }
);
Table.displayName = 'Table';
