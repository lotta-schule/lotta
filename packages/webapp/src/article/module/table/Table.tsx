import * as React from 'react';
import { ContentModuleModel } from 'model';
import { Edit } from './Edit';
import { Show } from './Show';

import styles from './Table.module.scss';

export interface TableCell {
    text: string;
}

export interface TableContent {
    rows: TableCell[][];
}

export interface TableConfiguration {}

export interface TableProps {
    contentModule: ContentModuleModel;
    isEditModeEnabled?: boolean;
    onUpdateModule?: (contentModule: ContentModuleModel) => void;
}

export const Table = React.memo<TableProps>(
    ({ isEditModeEnabled, contentModule, onUpdateModule }) => {
        return (
            <div className={styles.root} data-testid="TableContentModule">
                {isEditModeEnabled && onUpdateModule && (
                    <Edit
                        contentModule={contentModule}
                        onUpdateModule={onUpdateModule}
                    />
                )}
                {(!isEditModeEnabled || !onUpdateModule) && (
                    <Show contentModule={contentModule} />
                )}
            </div>
        );
    }
);
Table.displayName = 'Table';
