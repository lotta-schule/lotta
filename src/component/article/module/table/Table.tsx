import React, { memo } from 'react';
import { ContentModuleModel } from '../../../../model';
import { CardContent, makeStyles } from '@material-ui/core';
import { Edit } from './Edit';
import { Show } from './Show';

export interface TableCell {
    text: string;
}

export interface TableContent {
    rows: TableCell[][];
}

export interface TableConfiguration {}

export interface TableProps {
    contentModule: ContentModuleModel;
    isEditModeEnabled?: boolean;
    onUpdateModule?: (contentModule: ContentModuleModel) => void;
}

const useStyles = makeStyles((theme) => ({
    root: {
        paddingTop: 30,
        paddingBottom: 0,
        margin: '0 16.6%',
        overflow: 'auto',
        [theme.breakpoints.down('sm')]: {
            margin: 0,
        },
    },
}));

export const Table = memo<TableProps>(
    ({ isEditModeEnabled, contentModule, onUpdateModule }) => {
        const styles = useStyles();
        return (
            <CardContent
                className={styles.root}
                data-testid="TableContentModule"
            >
                {isEditModeEnabled && onUpdateModule && (
                    <Edit
                        contentModule={contentModule}
                        onUpdateModule={onUpdateModule}
                    />
                )}
                {(!isEditModeEnabled || !onUpdateModule) && (
                    <Show contentModule={contentModule} />
                )}
            </CardContent>
        );
    }
);
