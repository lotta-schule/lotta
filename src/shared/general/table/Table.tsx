import * as React from 'react';
import clsx from 'clsx';

import styles from './Table.module.scss';

export type TableProps = React.HTMLProps<HTMLTableElement>;

export const Table: React.FC<TableProps> = ({
    children,
    className,
    ...props
}) => {
    return (
        <table className={clsx(styles.root, className)} {...props}>
            {children}
        </table>
    );
};
