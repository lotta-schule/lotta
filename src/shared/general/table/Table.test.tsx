import { render } from '@testing-library/react';
import { Table } from './Table';

describe('Table', () => {
    it('should render', () => {
        const screen = render(
            <Table>
                <thead>
                    <tr>
                        <td>Header 1</td>
                        <td>Header 2</td>
                        <td>Header 3</td>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <td>Content 1</td>
                        <td>Content 2</td>
                        <td>Content 3</td>
                    </tr>
                    <tr>
                        <td>Content 1</td>
                        <td>Content 2</td>
                        <td>Content 3</td>
                    </tr>
                    <tr>
                        <td>Content 1</td>
                        <td>Content 2</td>
                        <td>Content 3</td>
                    </tr>
                </tbody>
            </Table>
        );
        expect(screen.container).toMatchSnapshot();
    });
});
