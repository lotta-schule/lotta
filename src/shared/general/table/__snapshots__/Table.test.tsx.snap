// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render 1`] = `
<div>
  <table
    class="root"
  >
    <thead>
      <tr>
        <td>
          Header 1
        </td>
        <td>
          Header 2
        </td>
        <td>
          Header 3
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Content 1
        </td>
        <td>
          Content 2
        </td>
        <td>
          Content 3
        </td>
      </tr>
      <tr>
        <td>
          Content 1
        </td>
        <td>
          Content 2
        </td>
        <td>
          Content 3
        </td>
      </tr>
      <tr>
        <td>
          Content 1
        </td>
        <td>
          Content 2
        </td>
        <td>
          Content 3
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
